test/unit/ssh-config.test.js

```diff
@@ -162,6 +162,22 @@ describe('SSHConfig', function() {
     assert.throws(function() { config.remove({}) })
   })
 
+  it('.remove by function', async function() {
+    const config = SSHConfig.parse(readFile('fixture/config'))
+    const length = config.length
+    
+    config.remove((line) => line.param && line.param.toLowerCase() === 'Host'.toLowerCase() && line.value === 'tahoe2')
+    assert(config.find({ Host: 'tahoe2' }) == null)
+    assert(config.length === length - 1)
+
+    config.remove((line) => line.param && line.param.toLowerCase() === 'HostName'.toLowerCase() && line.value === 'tahoe1.com')
```

Comment:
This one won't work because parameters behind `Host` or `Match` will be grouped as sub config, the structure would be like:

```js
[ { param: 'Host', value: 'tahoe1', 
     config: [ { param: 'HostName', value: 'tahoe1.com' }, ... ] } ]
```

Reply:
Yes my fault, I was doing similar tests on my project, but I was already in the config object. 
I remove the second `config.remove()` :)